scalableminds/webknossos
# Add possibility to move nodes

Add the possibility to move nodes by adding an action/reducer and by making the WebGL buffer updater (`skeleton.js`) update node and edge positions.
The active node can be moved using "Ctrl + Arrow Keys" or by using "Ctrl + LeftMouseDrag".

### URL of deployed dev instance (used for testing):
- https://movenodes.webknossos.xyz

### Steps to test:
- Open a tracing and move nodes using the mentioned keyboard and mouse controls.
  - Does it work in each viewport?
  - Is the node position persisted after a reload?
  - Does it work for branchpoints?

- Testing the merger mode:
  - open a skeleton tracing with a segmentation layer
  - enable the merger mode and create two nodes that merge to segments
  - Now move one of the nodes around. The mapping should update as expected.

### Issues:
- This will allow users to easily correct ground truth skeleton data. Previously one needed to remove a node, add a new node and reconnect the split trees.

------
